src/chocolatey.resources/redirects/RefreshEnv.cmd

```diff
@@ -11,53 +11,63 @@
 ::echo "RefreshEnv.cmd only works from cmd.exe, please install the Chocolatey Profile to take advantage of refreshenv from PowerShell"
 echo | set /p dummy="Refreshing environment variables from registry for cmd.exe. Please wait..."
 
+:: Generate unique file names
+for /F "tokens=* usebackq" %%F IN (`powershell -command "[guid]::NewGuid().ToString()"`) do set REFRESHENV_GUID=%%F

```

Comment:
This would require at least a 2 second start up time for PowerShell. Not that I'm against that, but that is longer than the entire thing takes to run now. So this is more than doubling the time. Are there any other methods?

Reply:
I'm afraid pure CMD doesn't provide anything reliable we could really use. Sounds an odd idea but what if we just implemented some tiny win32 application (with C) that would only print its process ID and that would be used as the random part of the temporary file names? The executable would be so small that it could be even put into version control beside RefreshEnv.cmd.

Reply:
LOL! 13 824 bytes :)

```C
#include <stdio.h>
#include <processthreadsapi.h>

void main()
{
    printf("%d", GetCurrentProcessId());
}
```

`gcc pid.c -o pid.exe -s -Os`